apps/snitch_core/lib/core/data/model/zone/country_zone.ex

```diff
@@ -0,0 +1,131 @@
+defmodule Snitch.Data.Model.CountryZone do
+  @moduledoc """
+  CountryZone API
+  """
+  use Snitch.Data.Model
+
+  import Ecto.Query
+
+  alias Ecto.Multi
+  alias Snitch.Data.Schema.{CountryZoneMember, Zone, Country}
+
+  @doc """
+  Creates a new state `Zone` whose members are `country_ids`.
+
+  `country_ids` is a list of primary keys of the `Snitch.Data.Schema.Country`s that
+  make up this zone. Duplicate IDs are ignored.
+  """
+  @spec create(String.t(), String.t(), [non_neg_integer]) :: term
+  def create(name, description, country_ids) do
+    zone_params = %{name: name, description: description, zone_type: "C"}
+    zone_changeset = Zone.changeset(%Zone{}, zone_params, :create)
+
+    Multi.new()
+    |> Multi.insert(:zone, zone_changeset)
+    |> Multi.run(:members, fn %{zone: zone} -> insert_members(country_ids, zone) end)
+    |> Repo.transaction()
+    |> case do
+      {:ok, %{zone: zone}} -> {:ok, zone}
+      error -> error
+    end
+  end
+
+  @spec delete(non_neg_integer | Zone.t()) ::
+          {:ok, Zone.t()} | {:error, Ecto.Changeset.t()} | {:error, :not_found}
+  def delete(id_or_instance) do
+    QH.delete(Zone, id_or_instance, Repo)
+  end
+
+  @spec get(map | non_neg_integer) :: Zone.t() | nil
+  def get(query_fields_or_primary_key) do
+    QH.get(Zone, query_fields_or_primary_key, Repo)
+  end
+
+  @spec get_all() :: [Zone.t()]
+  def get_all, do: Repo.all(from(z in Zone, where: z.zone_type == "S"))
```

Comment:
what is this `S` type zone and why default scope to it ?

Reply:
Good catch! This should have been `z.zone_type == "C"`.
> why default scope to it?
I'm sorry I don't understand what you mean.

Reply:
In rails we some time add default scope, which is like a default where query to get all the records in DB.

Here in this cases `z.zone_type == "C"` is a default scope to which changes the definition of get_all as in get all C type country.

Reply:
how about a diff name to function. Like `CountryZone.zone_type_c`

Reply:
Now got it. LGTM 👍